flood_adapt/adapter/fiat_adapter.py

```diff
@@ -53,27 +56,35 @@ def __init__(self, model_root: str, database_path: str) -> None:
             self.bfe["name"] = self.site.attrs.fiat.bfe.field_name
 
     def __del__(self) -> None:
-        for handler in self._logger.handlers:
+        for handler in self.logger.handlers:
             handler.close()
-        self._logger.handlers.clear()
+            self.logger.removeHandler(handler)
+        self.logger.handlers.clear()
         # Use garbage collector to ensure file handlers are properly cleaned up
         gc.collect()
 
-    def set_hazard(self, hazard: Hazard) -> None:
-        map_fn = hazard._get_flood_map_path()
-        map_type = hazard.site.attrs.fiat.floodmap_type
-        var = "zsmax" if hazard.event_mode == Mode.risk else "risk_maps"
-        is_risk = hazard.event_mode == Mode.risk
+    def __enter__(self) -> "FiatAdapter":
+        return self
 
-        # Add the hazard data to a data catalog with the unit conversion
-        wl_current_units = UnitfulLength(value=1.0, units="meters")
+    def __exit__(self, exc_type, exc_value, traceback) -> bool:
+        del self.fiat_model
+        del self
+
+        return False
```

Comment:
Its something we use for closing file handlers. We define a custom __del__ function for adapters that closes any files associated with it. We maybe should them them into a descriptive function or something.

But yeah deleting self is a bit weird. However since this happens either when exiting a `with FiatAdapter(...) as fa:` block, or when explicitely deleting the fiatadapter object at the end of a function to close all logfiles, I think it should be okay since we dont use it anymore after deleting.

But by all means I wouldnt mind a better solution